apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx

```diff
@@ -0,0 +1,185 @@
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import UserPreferences from '@cdo/apps/lib/util/UserPreferences';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import {default as LinkedButton} from '@cdo/apps/templates/Button';
+import {
+  setHasSeenProgressTableInvite,
+  setDateProgressTableInvitationDelayed,
+  setShowProgressTableV2,
+} from '@cdo/apps/templates/currentUserRedux';
+import i18n from '@cdo/locale';
+
+import styles from './progress-v2-invitation.module.scss';
+
+const newProgressViewGraphic = require('@cdo/static/teacherDashboard/progressOpenBetaAnnouncementGraphic.png');
+
+function InviteToV2ProgressModal({
+  sectionId,
+
+  // from redux
+  dateProgressTableInvitationDelayed,
+  hasSeenProgressTableInvite,
+  setHasSeenProgressTableInvite,
+  setShowProgressTableV2,
+  setDateProgressTableInvitationDelayed,
+}) {
+  const [invitationOpen, setInvitationOpen] = React.useState(false);
+
+  React.useEffect(() => {
+    const timeSinceInvitationLastDelayed = () => {
+      const startingDate = new Date(dateProgressTableInvitationDelayed);
+      const today = new Date();
+      const differenceInMilliseconds = today.getTime() - startingDate.getTime();
+      const differenceInDays = differenceInMilliseconds / (1000 * 3600 * 24);
```

Comment:
```suggestion
      const differenceInDays = differenceInMilliseconds / MILLISECONDS_IN_ONE_DAY;
```
Optional nit: move this to a constant with a meaningful name.